Automattic/wp-calypso
# Site Assembler: Get rendered patterns with the correct language

Related to https://github.com/Automattic/wp-calypso/issues/75421

## Proposed Changes

This PR is proposing to get the rendered patterns with a specific language via the changes below
* Pass `_locale=user` to the `/block-renderer/patterns/render` endpoint
* Fetch all patterns with the user locale instead of the site locale
* Use the `key` of the categories as the key of the category map instead of the slug since the slug will be translated
* Remove the `images` category from the `patternCategoriesOrder` to align the behavior in the site editor

https://user-images.githubusercontent.com/13596067/230887486-e11d200f-900e-48f8-b6fc-ebe857f86ae7.mov

## Testing Instructions



* Apply D107414-code to your sandbox
* Switch the language of your account to non-English
* Go to `/setup?siteSlug=<your_site>`
  * Note that the site should be the site with a free plan
* Continue until you land on the Design Picker
* On the Design Picker screen, scroll to bottom and select "Start designing"
* On the Pattern Assembler screen
  * Select `Header`, and ensure the patterns are translated
  * Select `Footer`, and ensure the patterns are translated
  * Select `Sections > Add patterns`
    * Select any category
    * Ensure the patterns are translated

## Pre-merge Checklist
